scripts/jobs/academy_data/load_all_academy_data_into_redshift.py

```diff
@@ -0,0 +1,143 @@
+    year, month, day = today.year, str(today.month).zfill(2), str(today.day).zfill(2)
+    return f"{base_s3_url}{glue_table_name}/import_year={year}/import_month={month}/import_day={day}/import_date={year}{month}{day}/"
+
+def process_load_tables(schema: str, catalog: str, table_mapping: Dict[str, str], iam_role: str, base_s3_url: str) -> None:
+    """Main function to process and load tables."""
+    creds = get_secret('data-and-insight', 'eu-west-2') # this need to change to the new role "database_migration" access id and access key;
+    glue_client = boto3.client('glue', region_name='eu-west-2', aws_access_key_id=creds['accessKeyId'], aws_secret_access_key=creds['secretAccessKey'])
+
+    for original_pattern, new_prefix in table_mapping.items():
+        for table in get_all_tables(glue_client, catalog, original_pattern):
+            new_table_name = table['Name'].replace(original_pattern, new_prefix)
+            create_sql = create_table_sql(schema, new_table_name, table['StorageDescriptor']['Columns'])
+            # print(create_sql)  #  print statements are for debugging
+
+            try:
+                rs_command(create_sql, fetch_results=False, allow_commit=True)
+                print(f"Table {schema}.{new_table_name} created successfully.")
+            except Exception as e:
+                if 'already exists' in str(e):
+                    print(f"Table {new_table_name} already exists.")
+                else:
+                    print(f"Failed to create table {new_table_name}: {e}")
+                    raise e
+
+            truncate_sql = truncate_table(schema, new_table_name)
+            rs_command(truncate_sql, fetch_results=False, allow_commit=True)
+            print(f"Table {schema}.{new_table_name} truncated successfully.")
+
+            s3_location = get_s3_location(table['Name'], base_s3_url)
+            copy_sql = copy_command(schema, new_table_name, s3_location, iam_role)
+            # print(copy_sql) #  print statements are for debugging
+            try:
+                rs_command(copy_sql, fetch_results=False, allow_commit=True)
+                print(f"Data for {datetime.today().strftime('%Y%m%d')} copied successfully into table {new_table_name}.")
+            except Exception as e:
+                print(f"Failed to copy data into table {schema}.{new_table_name}: {e}")
+
+            time.sleep(1)  # Sleep to mitigate risk of overwhelming the cluster
+
+def main():
+    # for all tables under ctax
+    process_load_tables(
+        schema='ctax', # Redshift schema
+        catalog='revenues-raw-zone', # Glue catalog database
+        # map the table names in Glue catalog to names in Redshift
+        table_mapping= {
+            'lbhaliverbviews_core_ct': 'ct',
+            'lbhaliverbviews_core_sy': 'sy'
+        },
+        iam_role='arn:aws:iam::120038763019:role/dataplatform-stg-redshift-serverless-role',
+        base_s3_url = "s3://dataplatform-stg-raw-zone/revenues/"
```

Comment:
it might be a good idea to import these as environment variables - you'll need them to change depending on where your script is deployed anyway.